bigchaindb/backend/README.md

```diff
@@ -0,0 +1,43 @@
+# Backend Interfaces
+
+## Structure
+
+- [`changefeed.py`](./changefeed.py): Changefeed-related interfaces
+- [`connection.py`](./connection.py): Database connection-related interfaces
+- [`query.py`](./query.py): Database query-related interfaces, dispatched through single-dispatch
+- [`schema.py`](./schema.py): Database setup and schema-related interfaces, dispatched through
+  single-dispatch
+
+Built-in implementations (e.g. [RethinkDB's](./rethinkdb)) are provided in sub-directories and
+have their connection type's location exposed as `BACKENDS` in [`connection.py`](./connection.py).
+
+## Single-Dispatch
```

Comment:
I've reworded the section: https://github.com/bigchaindb/bigchaindb/pull/918/commits/8d903bc459e5225b128a9a4323fb761350c92d6b.